lib/mayaUsd/fileio/jobs/readJob.cpp

```diff
@@ -141,7 +141,11 @@ bool UsdMaya_ReadJob::Read(std::vector<MDagPath>* addedDagPaths)
     } else {
         UsdStageCacheContext stageCacheContext(UsdMayaStageCache::Get(
             mImportData.stageInitialLoadSet() == UsdStage::InitialLoadSet::LoadAll));
-        stage = UsdStage::Open(rootLayer, sessionLayer, mImportData.stageInitialLoadSet());
+
+        if (mArgs.pullImport)
```

Comment:
Can you add a code comment here describing why a pull import is different from a regular import?

Reply:
So does a pullImport bring everything in "fresh"? and create new reference nodes? always or just in cases where the target refNode isn't named exactly right? (if we ignore the session layer, we won't have the mapping between prim and maya reference node)

I wonder if adding a "why" it ignores the session layer here would help it be clear.

Reply:
Stage search in cache doesn't affect how the mayaReferenceNodeName attribute is used in sessionLayer to match reference node. New reference node are created only if no match with mayaReferenceNodeName and if the expected name is not found in all the non-connected reference node.